src/blog/2024/04/using-webhook-with-node-red.md

```diff
@@ -0,0 +1,184 @@
+
+## How Webhook works 
+
+!["Image displaying how webhook works"](./images/using-webhook-with-node-red-how-webhook-works.png "Image displaying how webhook works"){data-zoomable}
+
+- Event Initiator: This refers to the event specified to trigger the WebHook. Whenever this event occurs, the WebHook will be triggered.
+- Webhook Server: The webhook server is responsible for managing webhook configurations and endpoints. It listens for the specified event. When the event is detected, the webhook server automatically sends an HTTP POST request containing relevant data to the designated third-party application or service.
+- Data Reception by Third-Party Application: The third-party application will receive the data sent via the WebHook to the designated URL or listener provided during registration.
+- Custom Action Execution: Upon receiving the POST request, specific actions can be performed.
+
+## API Vs Webhook
+
+It's common and understandable to get confused between APIs and webhooks, especially when you are learning about webhooks for the first time. However, comparing the two can help dispel these confusions.
+
+| Aspect            | API                                                    | Webhook                                              |
+|-------------------|--------------------------------------------------------|------------------------------------------------------|
+| Direction         | Typically involves client-to-server communication.     | Typically involves server-to-server communication.   |
+| Initiation        | The client initiates requests.                  | The server initiates requests.                |
+| Request Method    | Usually employs HTTP methods like GET, POST, etc.      | Typically uses the HTTP POST method.                 |
+| Response          | Provides an immediate response upon request.           | Does not provide an immediate response; asynchronous.|
+| Data Transfer     | Utilizes a pull model where the client fetches data.  | Operates on a push model where the server pushes data to the client.|
+| Polling           | Requires periodic polling for updates.                 | No need for polling; receives updates directly.      |
+| Payload           | The client specifies the payload in the request.       | The server defines the payload in the outgoing request. |
+| Error Handling    | Typically includes error codes and messages.           | Errors are handled by retry mechanisms or manual intervention. |
+
+### Example Scenario:
+
+Consider a manufacturing facility that utilizes temperature sensors to monitor temperature levels in critical areas. When the temperature falls or exceeds predefined thresholds, it triggers a series of actions for maintenance and monitoring.
+
+!["Diagram explaining how component works in Webhook"](./images/using-webhook-with-node-red-diagram.png "Diagram explaining how component works in Webhook"){data-zoomable}
```

Comment:
@ZJvandeWeg I have also updated the diagram. Could you please check it and let me know if it is correct?